src/models/poll.ts

```diff
@@ -189,10 +189,14 @@ export class Poll extends TypedEventEmitter<Exclude<PollEvent, PollEvent.New>, P
         /**
          * MSC3381
          * If a m.poll.end event is received from someone other than the poll creator or user with permission to redact
-         * other's messages in the room, the event must be ignored by clients due to being invalid.
+         * others' messages in the room, the event must be ignored by clients due to being invalid.
          */
         const roomCurrentState = this.room.currentState;
         const endEventSender = endEvent.getSender();
-        return !!endEventSender && roomCurrentState.maySendRedactionForEvent(this.rootEvent, endEventSender);
+        return (
+            !!endEventSender &&
+            (endEventSender === this.matrixClient.getSafeUserId() ||
```

Comment:
this is checking if the sender of the end event is the *current* user? Don't we want to check if it was the sender of the `rootEvent`?

```suggestion
            (endEventSender === this.rootEvent.getSender() ||
```

Reply:
🤦‍♀️ fixed and improved the tests to catch this. Thanks!